hack/lib/golang.sh

```diff
@@ -785,6 +788,18 @@ kube::golang::build_binaries() {
     goasmflags="-trimpath=${KUBE_ROOT}"
     gogcflags="${GOGCFLAGS:-} -trimpath=${KUBE_ROOT}"
 
+    # extract tags if any specified in GOFLAGS
+    # shellcheck disable=SC2001
+    gotags=$(echo "${GOFLAGS:-}" | sed -e 's|.*-tags=\([^-]*\).*|\1|')
```

Comment:
I think this doesn't work correctly for multiple tags.
If I want to set multiple tags via `GOFLAGS` to go I have to use:
`export GOFLAGS='-tags=selinux -tags=providerless'`

this command will give me:
```console
echo "${GOFLAGS:-}" | sed -e 's|.*-tags=\([^-]*\).*|\1|'
providerless
```

Reply:
something like the above edit, I'm sure it can be done more cleanly ... 🤔

Reply:
Thankfully I think `GOFLAGS` specifically expects a space separated set of `-tag=value` so there may not be too many other edge cases...

```
GOFLAGS

	A space-separated list of -flag=value settings to apply
	to go commands by default, when the given flag is known by
	the current command. Each entry must be a standalone flag.
	Because the entries are space-separated, flag values must
	not contain spaces. Flags listed on the command line
	are applied after this list and therefore override it.
```

https://golang.org/cmd/go/#hdr-Environment_variables

I am not sure if we even need the quote stripping.

Reply:
@BenTheElder from what i understood `-tags=selinux -tags=providerless` the last one is picked up. So if you do this today `selinux` would not kick in.

So the test cases i had were as follows (quoting from memory):
- `-tags=providerless`
- `-tags='providerless'`
- `-tags="providerless"`
- `-abc -tags=providerless`
- `-abc -tags="providerless"`
- `-abc -tags='providerless'`
- `-abc -tags=providerless -xyz`
- `-abc -tags="providerless" -xyz`
- `-abc -tags='providerless' -xyz`
- `-abc -tags="providerless selinux" -xyz`
- `-abc -tags='providerless selinux' -xyz`

Reply:
So GOFLAGS cannot contain quoted values and the value in -flag=value cannot contain a space.

This means this flag couldn't have more than one value:
https://github.com/golang/go/issues/26849

Except in go1.13 it may be comma seperated for cmd/go
https://github.com/golang/go/commit/80e7832733fd245181af3394077f2df21303a4aa

Need to do some tests when I get to a machine.

Reply:
Ack. please let me know. Let's wrap this up with spaces first please as the first iteration. i can follow up with support for comma.

Reply:
```
[bentheelder@bentheelder:~/junk·2020-01-31T12:31:41-0800]
$ GOFLAGS='-tags=providerless,selinux' go build .

[bentheelder@bentheelder:~/junk·2020-01-31T12:31:48-0800]
$ cat ./a.go
// +build providerless,selinux

package main

func main() {
	println("test")
}
```

Reply:
versus:
```
[bentheelder@bentheelder:~/junk·2020-01-31T12:32:03-0800]
$ GOFLAGS='-tags="providerless selinux"' go build .
go: parsing $GOFLAGS: non-flag "selinux\""

```

Reply:
we can replace all of this processing with just:
`gotags="selinux,$(echo "${GOFLAGS:-}" | sed -e 's|.*-tags=\([^-]*\).*|\1|')"`